diff --git a/cmake/SuperBuild.cmake b/cmake/SuperBuild.cmake
index fa3ead64..1a57cdcd 100644
--- a/cmake/SuperBuild.cmake
+++ b/cmake/SuperBuild.cmake
@@ -351,10 +351,13 @@ if("${BUILD_TESTING}")
 	"set(ADDPATH [=[${CMAKE_BINARY_DIR}/staging/${CMAKE_INSTALL_LIBDIR}:${CMAKE_BINARY_DIR}/staging/lib]=])\n"
 	[===[
 set(LD_LIBRARY_PATH "$ENV{LD_LIBRARY_PATH}")
+set(DYLD_LIBRARY_PATH "$ENV{DYLD_LIBRARY_PATH}")
 if("x${LD_LIBRARY_PATH}x" STREQUAL xx)
 	set(ENV{LD_LIBRARY_PATH} "${ADDPATH}")
+	set(ENV{DYLD_LIBRARY_PATH} "${ADDPATH}")
 else()
 	set(ENV{LD_LIBRARY_PATH} "${ADDPATH}:${LD_LIBRARY_PATH}")
+	set(ENV{DYLD_LIBRARY_PATH} "${ADDPATH}:${DYLD_LIBRARY_PATH}")
 endif()
 ]===]
 	)
