diff --git a/pdi/src/expression/impl.cxx b/pdi/src/expression/impl.cxx
index 486cf5e6..4406b4ec 100644
--- a/pdi/src/expression/impl.cxx
+++ b/pdi/src/expression/impl.cxx
@@ -80,9 +80,18 @@ string Expression::Impl::to_string(Context& ctx) const
 
 Ref Expression::Impl::to_ref(Context& ctx, Datatype_sptr type) const
 {
-	Ref_rw result{aligned_alloc(type->alignment(), type->buffersize()), [](void* v) { free(v); }, type, true, true};
-	copy_value(ctx, result.get(), type);
-	return result;
+	auto data = std::aligned_alloc(type->alignment(), type->buffersize());
+	if (data) {
+		Ref_rw result{data, [](void* v) { free(v); }, type, true, true};
+		copy_value(ctx, result.get(), type);
+		return result;
+	} else {
+		auto al = static_cast<std::align_val_t>(type->alignment());
+		data = operator new (type->buffersize(), al);
+		Ref_rw result{data, [al](void* v) { operator delete (v, al); }, type, true, true};
+		copy_value(ctx, result.get(), type);
+		return result;
+	}
 }
 
 unique_ptr<Expression::Impl> Expression::Impl::parse(PC_tree_t value)
