diff --git a/pdi/tests/operators.h b/pdi/tests/operators.h
index b9133ac9..93b949fa 100644
--- a/pdi/tests/operators.h
+++ b/pdi/tests/operators.h
@@ -180,10 +180,18 @@ inline constexpr auto ALL_VALS<int8_t>
        std::numeric_limits<int8_t>::max()};
 
 template <>
-inline constexpr auto ALL_VALS<uint64_t> = {std::numeric_limits<uint64_t>::min() + 1, -1ul, 1ul, std::numeric_limits<uint64_t>::max()};
+inline constexpr auto ALL_VALS<uint64_t>
+	= {static_cast<uint64_t>(std::numeric_limits<uint64_t>::min() + 1),
+       static_cast<uint64_t>(-1),
+       static_cast<uint64_t>(1),
+       std::numeric_limits<uint64_t>::max()};
 
 template <>
-inline constexpr auto ALL_VALS< int64_t> = {std::numeric_limits<int64_t>::min() + 1, -1l, 1l, std::numeric_limits<int64_t>::max()};
+inline constexpr auto ALL_VALS<int64_t>
+	= {static_cast<int64_t>(std::numeric_limits<int64_t>::min() + 1),
+       static_cast<int64_t>(-1),
+       static_cast<int64_t>(1),
+       std::numeric_limits<int64_t>::max()};
 
 template <>
 inline constexpr auto ALL_VALS<float> = {std::numeric_limits<float>::min(), -1.5f, -1.f, -.5f, .5f, 1.f, 1.5f, std::numeric_limits<float>::max()};
